addons/l10n_eu_service/models/res_company.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from .eu_tax_map import EU_TAX_MAP
+
+
+class Company(models.Model):
+    _inherit = 'res.company'
+
+    @api.model
+    def map_all_company_taxes(self):
+        if not self.env['ir.config_parameter'].sudo().get_param('l10n_eu_service.taxes_mapped', False):
+            eu_countries = self.env.ref('base.europe').country_ids
+            companies = self.search([]).filtered(lambda c: c.country_id in eu_countries)
+            companies.map_eu_taxes()
+            self.env['ir.config_parameter'].sudo().set_param('l10n_eu_service.taxes_mapped', True)
+
+    def map_eu_taxes(self):
+        eu_countries = self.env.ref('base.europe').country_ids
+        tax_groups = {}
+        for company in self:
+            invoice_repartition_lines, refund_repartition_lines = company._get_repartition_lines_oss()
+            taxes = self.env['account.tax'].search([('type_tax_use', '=', 'sale'),
+                ('amount_type', '=', 'percent'),
+                ('company_id','=', company.id),
+                ('tax_group_id', 'not ilike', 'OSS')])
```

Comment:
@oco-odoo due to this additional domain filter, we can remove the `ir.config.parameter` and it should be safe to run `map_all_company_taxes` a second time.
So users can 'refresh' the Taxes and FPs by updating the module

Reply:
I don't think we need this if we apply my suggestion about existing mappings. What's mapped won't be modified, so re-updating the module should have no effect and we can indeed remove the config parameter.

Reply:
There will be new taxes created - so reinstalling the module will map the mapped taxes ... :)
the OSS domestic tax won't have an existing mapping either and will pass the check: `if tax_amount and not domestic_tax in fpos.tax_ids:`

Reply:
storing 'account.tax.group' references in `ir.model.data` so the name search is removed